pex/pex.py

```diff
@@ -68,11 +68,15 @@ def _activate(self):
 
       # set up the local .pex environment
       pex_info = self._pex_info.copy()
+      pex_info_pex_path = pex_info.pex_path
+      overrides_pex_path = self._pex_info_overrides.pex_path
       pex_info.update(self._pex_info_overrides)
```

Comment:
I assume that you found that this `pex_info.update()` call does not actually combine env vars + PEX-INFO, but instead clobbers one or the other?

Reply:
Yup! Env vars clobber the existing vars set by pex info.